### PR TITLE
Support Durango Upgrade

### PR DESCRIPTION
**Pull request by dhrubabasu**
* Adds construction of `AddPermissionlessValidatorTx` and `AddPermissionlessDelegatorTx`
* Removes `Memo` field population from staking transactions
* Adds parsing support for `BaseTx` and `TransferSubnetOwnershipTx`
* Adds TODOs for future cleanups

### CODE REVIEW

File: mapper/pchain/tx_builder.go [hunk 1]
```diff
@@ -218,31 +320,95 @@ func buildAddDelegatorTx(
...
+		return nil, nil, fmt.Errorf("parse outputs failed: %w", err)
 	}
 
 	weight, err := sumOutputAmounts(stakeOutputs)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	tx := &txs.Tx{Unsigned: &txs.AddDelegatorTx{
+	subnetBytes, err := mapper.DecodeToBytes(metadata.Subnet)
```
Comment (bilgehansahin) [inline mapper/pchain/tx_builder.go:389]: Can we default this to `constants.PrimaryNetworkID` if `metadata.Subnet` is undefined / empty? For the time being, primary network is the only one that we are supporting, and I'd prefer not to hardcode empty ids in the client side if possible.

**Review by bilgehansahin — COMMENTED** [review]
Not a blocker, just a suggestion. You can also test these transactions ad-hoc by adding them to the postman collection in `postman/Avalanche%20Rosetta.postman_collection.json`. You can clone AddDelegator/AddValidator folders under `Construction/P Chain` and modify the `/construction/preprocess` input. The rest is automatically wired.

File: service/backend/pchain/construction.go
Comment (bilgehansahin) [inline service/backend/pchain/construction.go:1]: You should also add the new construction types to line 72 in `ConstructionMetadata` method.

**Review by bilgehansahin — APPROVED** [review]

File: mapper/pchain/tx_parser.go [hunk 2]
```diff
@@ -264,7 +272,7 @@ func (t *TxParser) parseAddValidatorTx(txID ids.ID, tx *txs.AddValidatorTx) (*tx
 }
 
 func (t *TxParser) parseAddPermissionlessValidatorTx(txID ids.ID, tx *txs.AddPermissionlessValidatorTx) (*txOps, error) {
-	ops, err := t.baseTxToCombinedOperations(txID, &tx.BaseTx, OpAddValidator)
+	ops, err := t.baseTxToCombinedOperations(txID, &tx.BaseTx, OpAddPermissionlessValidator)
```
Comment (bilgehansahin) [inline mapper/pchain/tx_parser.go:275]: Can you also append the `validator_rewards_owner` / `delegator_rewards_owner` / `subnet_id` in `addMetadataToStakeOuts` below for the new transactions?

**Review by bilgehansahin — APPROVED** [review]

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**